convert.go

```diff
@@ -68,40 +74,83 @@ func InterfaceToVariable(input interface{}) (ast.Variable, error) {
 	var mapVal map[string]interface{}
 	if err := hilMapstructureWeakDecode(input, &mapVal); err == nil {
 		elements := make(map[string]ast.Variable)
+		elemType := ast.TypeInvalid
 		for i, element := range mapVal {
 			varElement, err := InterfaceToVariable(element)
 			if err != nil {
 				return ast.Variable{}, err
 			}
 			elements[i] = varElement
+			if elemType == ast.TypeInvalid {
+				elemType = varElement.Type
+			} else if elemType != varElement.Type {
```

Comment:
This also needs to handle `TypeUnknown` as a special case, so that computed values can pass through.